apps/wallet/src/dapp-interface/WalletStandardInterface.ts

```
@@ -0,0 +1,185 @@
+            },
+            'sui:signAndExecuteTransaction': {
+                version: '1.0.0',
+                signAndExecuteTransaction: this.#signAndExecuteTransaction,
+            },
+        };
+    }
+
+    get accounts() {
+        return this.#account ? [this.#account] : [];
+    }
+
+    constructor() {
+        this.#events = mitt();
+        this.#account = null;
+        this.#messagesStream = new WindowMessageStream(
+            'sui_in-page',
+            'sui_content-script'
+        );
+
+        this.#connected();
+    }
+
+    on<E extends WalletEventNames>(
+        event: E,
+        listener: WalletEvents[E]
+    ): () => void {
+        this.#events.on('standard:change', listener);
+        return () => this.#events.off(event, listener);
+    }
+
+    #connected = async () => {
+        const accounts = await mapToPromise(
+            this.#send<GetAccount, GetAccountResponse>({
+                type: 'get-account',
+            }),
+            (response) => response.accounts
+        );
+
+        const properties: ('accounts' | 'chains')[] = [];
+
+        const [address] = accounts;
+
+        if (address) {
+            const account = this.#account;
+            if (!account || account.address !== address) {
+                this.#account = new ReadonlyWalletAccount({
+                    address,
+                    // TODO: Expose public key instead of address:
+                    publicKey: new Uint8Array(),
+                    chains: SUI_CHAINS,
+                    features: [
+                        'sui:signAndExecuteTransaction',
+                        'standard:signMessage',
+                    ],
+                });
+                properties.push('accounts');
+            }
+        }
+
+        if (properties.length) {
+            this.#events.emit('standard:change', properties);
+        }
+    };
+
+    #connect: ConnectMethod = async (input) => {
+        if (!input?.silent) {
```

I don't understand why we need outside to pass `slient` to tell if the permission prompt should popup.
It should be judged by ext checking the permission approval history.
What use case would there be?

---

The intent here is for auto-connecting to the users preferred wallet. You don’t want to prompt the user for connection, but if the permissions have already been granted then it can be connected to automatically.

---

What about this case: if the `slient` passed in but no permission granted, wallet exts still need to prompt for approval

---

The intent there is actually that the connect would just silently fail, without prompting the user.

---

This could be a bug from user's perspective? Not convinced. More use cases?

---

It couldn't be perceived be a bug because this is only for the auto-connect flow. The user will just see a button to connect their account as if it was not connected.

---

Alright, then we follow your logic and try